### PR TITLE
Enable reader / writer to use TF numbers consistently in input / output

### PR DESCRIPTION
**Pull request by pbuehler**

### CODE REVIEW

**Comment by pbuehler** [comment]
@jgrosseo this PR allows the writer to access the TF numbers read by the reader from the input files and to use them to form the output folder names accordingly.

**Review by jgrosseo — DISMISSED** [review]

**Comment by jgrosseo** [comment]
@ktf the error seems unrelated "killed due to memory"

File: Framework/Core/include/Framework/DataInputDirector.h [hunk 1]
```diff
@@ -103,10 +106,12 @@ struct DataInputDirector {
 
   // getters
   DataInputDescriptor* getDataInputDescriptor(header::DataHeader dh);
+  int getNumberInputDescriptors() { return mdataInputDescriptors.size(); }
+
   std::unique_ptr<TTreeReader> getTreeReader(header::DataHeader dh, int counter, int numTF, std::string treeName);
-  std::tuple<TFile*, std::string> getFileFolder(header::DataHeader dh, int counter, int numTF);
   TTree* getDataTree(header::DataHeader dh, int counter, int numTF);
-  int getNumberInputDescriptors() { return mdataInputDescriptors.size(); }
+  int64_t getTimeFrameNumber(header::DataHeader dh, int counter, int numTF);
```
Comment (ktf) [inline Framework/Core/include/Framework/DataInputDirector.h]: Please make sure signed / unsigned match. I think you use unsigned above.

File: Framework/Core/include/Framework/DataInputDirector.h [hunk 2]
```diff
@@ -103,10 +106,12 @@ struct DataInputDirector {
...
+  std::tuple<TFile*, std::string> getFileFolder(header::DataHeader dh, int counter, int numTF);
```
Comment (ktf) [inline Framework/Core/include/Framework/DataInputDirector.h]: Please do not use std::tuple, but create a small struct with meaningful names for the members (no need to have getters).

File: Framework/Core/src/CommonDataProcessors.cxx [hunk 3]
```diff
@@ -399,32 +399,57 @@ DataProcessorSpec
     auto& callbacks = ic.services().get<CallbackService>();
     callbacks.set(CallbackService::Id::EndOfStream, endofdatacb);
 
+    // prepare map<uint64_t, uint64_t>(startTime, TFNumber)
+    std::map<uint64_t, uint64_t> TFNumbers;
```
Comment (ktf) [inline Framework/Core/src/CommonDataProcessors.cxx]: please make sure variables always start with lowercase letters.  I would use `tfNumbers` or `timeframeNumbers` in this particular case.

File: Framework/Core/src/CommonDataProcessors.cxx [hunk 4]
```diff
@@ -399,32 +399,57 @@ DataProcessorSpec
...
     // this functor is called once per time frame
-    return std::move([dod](ProcessingContext& pc) mutable -> void {
+    return std::move([dod, TFNumbers](ProcessingContext& pc) mutable -> void {
       LOGP(DEBUG, "======== getGlobalAODSink::processing ==========");
       LOGP(DEBUG, " processing data set with {} entries", pc.inputs().size());
 
       // return immediately if pc.inputs() is empty. This should never happen!
-      auto ninputs = pc.inputs().size();
-      if (ninputs == 0) {
+      if (pc.inputs().size() == 0) {
```
Comment (ktf) [inline Framework/Core/src/CommonDataProcessors.cxx:411]: .empty()

File: Framework/Core/include/Framework/DataInputDirector.h (hunk 1 above)
Comment (pbuehler) [reply Framework/Core/include/Framework/DataInputDirector.h]: done

File: Framework/Core/include/Framework/DataInputDirector.h (hunk 2 above)
Comment (pbuehler) [reply Framework/Core/include/Framework/DataInputDirector.h]: done

File: Framework/Core/src/CommonDataProcessors.cxx (hunk 3 above)
Comment (pbuehler) [reply Framework/Core/src/CommonDataProcessors.cxx]: done

File: Framework/Core/src/CommonDataProcessors.cxx (hunk 4 above)
Comment (pbuehler) [reply Framework/Core/src/CommonDataProcessors.cxx:411]: > .empty()
‘class o2::framework::InputRecord’ has no member named ‘empty’

Comment (ktf) [reply Framework/Core/src/CommonDataProcessors.cxx:411]: ah, ok, right.

**Pull request merged**